src/components/AssetsStatusAlert/index.jsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StatusAlert } from '@edx/paragon';
+import { FormattedNumber } from 'react-intl';
+
+import { assetActions } from '../../data/constants/actionTypes';
+import WrappedMessage from '../../utils/i18n/formattedMessageWrapper';
+import messages from './displayMessages';
+
+
+export default class AssetsStatusAlert extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      statusAlertFields: {
+        alertDialog: '',
+        alertType: 'info',
+      },
+      statusAlertOpen: false,
+      uploadSuccessCount: 1,
+    };
+
+    this.statusAlertRef = {};
+
+    this.closeStatusAlert = this.closeStatusAlert.bind(this);
+    this.updateStatusAlertFields = this.updateStatusAlertFields.bind(this);
+    this.updateUploadSuccessCount = this.updateUploadSuccessCount.bind(this);
+
+    this.closeDeleteStatus = this.closeDeleteStatus.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { assetsStatus, deletedAsset } = nextProps;
+    this.updateStatusAlertFields(assetsStatus, deletedAsset);
+  }
+
+  updateStatusAlertFields(assetsStatus, deletedAsset) {
+    const assetName = deletedAsset.display_name;
+    let alertDialog;
+    let alertType;
+
+    const genericUpdateError = (
+      <WrappedMessage
+        message={messages.assetsStatusAlertGenericUpdateError}
+      />
+    );
+
+    switch (assetsStatus.type) {
```

Comment:
It feels like this `switch` could be moved out into it's own function since what it's really doing is generating two pieces of information that will eventually get used in the `statusAlertFields` update?

Maybe something like

```javascript
getStatusAlertFields(assetStatus) {
  // add switch here
  ...
  return { alertDialog, alertType }
}

updateStatusAlertFields(assetsStatus, deletedAsset) {
  this.setState({
    statusAlertOpen: true,
    statusAlertFields: this.getStatusAlertFields(assetStatus),
  });
}
```